### PR TITLE
feat(API): JIRA-5751 Add API provider

### PR DESCRIPTION
**Pull request by lukasjuhas**
This is related to an internal task I've bee working on. Part of the logic, is being moved to the package instead.

### CODE REVIEW

**Comment by lukasjuhas** [comment]
Fake provider and test are WIP. Coming shortly.

**Review by owenvoke — COMMENTED** [review]
Looking good, nice and clean! I just have a few small comments. 👍🏻

File: src/Traits/InteractsWithFeatureFlags.php [hunk 1]
```diff
@@ -8,18 +8,16 @@ trait InteractsWithFeatureFlags
 {
     public function switchFeatureFlag(string $key, bool $onOff): void
     {
-        $this->app['config']->set('feature-flags.overrides.'.$key, $onOff);
+        $this->app['config']->set(sprintf('feature-flags.overrides.%s', $key), $onOff);
```
[the remark this answers is not in the document]
Comment (lukasjuhas) [reply src/Traits/InteractsWithFeatureFlags.php]: I agree, this is much better.

**Review by owenvoke — APPROVED** [review]

**Comment by owenvoke** [comment]
We can ignore the Windows test failure, it's an issue with Windows only, and doesn't reflect actual conditions. 😬

**Pull request merged**